core/cont/src/TClassTable.cxx

```diff
@@ -809,9 +894,12 @@ void ROOT::AddClassAlternate(const char *normName, const char *alternate)
 
 void ROOT::ResetClassVersion(TClass *cl, const char *cname, Short_t newid)
 {
-   if (cname && cname!=(void*)-1) {
-      TClassRec *r = TClassTable::FindElement(cname,kFALSE);
-      if (r) r->fId = newid;
+
+   if (cname && cname != (void*)-1 && TClassTable::CheckClassTableInit()) {
```

Comment:
Regarding the `cname != (void*)-1`: this was there before, and seems to be also a special case below.  Maybe the documentation for this function also has to mention that?

Reply:
Isn't it documented line 885?

Reply:
The documentation (or rather the function interface) is not probably what I would call "friendly"; but that's unrelated to this PR anyway.